wasm/src/main/scala/ir2wasm/HelperFunctions.scala

```diff
@@ -836,4 +837,70 @@ object HelperFunctions {
     }
   }
 
+  def genNewDefault(clazz: LinkedClass)(implicit ctx: TypeDefinableWasmContext): Unit = {
+    val className = clazz.name.name
+    val classInfo = ctx.getClassInfo(className)
+    assert(clazz.hasDirectInstances)
+
+    val structName = WasmTypeName.WasmStructTypeName(className)
+    val fctx = WasmFunctionContext(
+      Names.WasmFunctionName.newDefault(className),
+      Nil,
+      List(WasmRefType(WasmHeapType.Type(structName)))
+    )
+
+    import fctx.instrs
+
+    val interfaces = classInfo.ancestors.map(ctx.getClassInfo(_)).filter(_.isInterface)
+
+    if (!interfaces.isEmpty) {
+
+      // check if itable is already initialized
+      val headIfaceIdx = ctx.getItableIdx(interfaces.head.name)
+      instrs += GLOBAL_GET(
+        WasmImmediate.GlobalIdx(WasmGlobalName.WasmGlobalITableName(className))
+      )
+      instrs += I32_CONST(WasmImmediate.I32(headIfaceIdx))
+      instrs += ARRAY_GET(WasmImmediate.TypeIdx(WasmTypeName.WasmArrayTypeName.itables))
+      instrs += REF_IS_NULL
```

Comment:
This is pretty expensive to do on every object instantiation. Consider moving "Fill in itables" to the `start` function, so that we don't clutter `newDefault` helpers with these tests.

Reply:
I'm thinking about we can test if the itable is already initialized by checking if the global itable is null, but yeah doing so in `start` function seems even better :+1: